src/test/java/org/agrona/agent/BufferAlignmentAgentTest.java

```diff
@@ -0,0 +1,320 @@
+package org.agrona.agent;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.function.IntConsumer;
+
+import org.agrona.BitUtil;
+import org.agrona.DirectBuffer;
+import org.agrona.ExpandableArrayBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.AtomicBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import net.bytebuddy.agent.ByteBuddyAgent;
+
+public class BufferAlignmentAgentTest
+{
+
+    private static final String TEST_STRING = "BufferAlignmentTest";
+
+    @BeforeClass
+    public static void installAgent()
+    {
+        BufferAlignmentAgent.agentmain("", ByteBuddyAgent.install());
+    }
+
+    @AfterClass
+    public static void removeAgent()
+    {
+        BufferAlignmentAgent.removeTransformer();
+    }
+
+    @Test
+    public void testUnsafeBufferFromByteArray()
+    {
+        testUnsafeBuffer(new UnsafeBuffer(new byte[256]), 0);
+    }
+
+    @Test
+    public void testUnsafeBufferFromByteArrayWithOffset()
+    {
+        final UnsafeBuffer buffer = new UnsafeBuffer(new byte[256], 1, 128);

```

Comment:
On 32bit JVMs this tends to fail. If you use ByteBuffer.allocateDirect() then is will be 8 byte aligned.